src/html_json.js

```diff
@@ -48,66 +46,18 @@ const helpers = {
   },
 };
 
-/**
- * Load an index configuration from a local file. Its location is given by the path
- * being requested (e.g. '/test/specs/blog/post.html').
- *
- * @param {object} params
- * @returns configuration
- */
-async function loadConfigFromFile(params) {
-  const { path, logger } = params;
-
-  const configfile = resolve(dirname(__dirname), dirname(path).substr(1), 'helix-index.yaml');
-  logger.debug(`Reading index configuration from: ${configfile}`);
-
-  try {
-    const source = await fs.readFile(configfile, 'utf8');
-    const document = YAML.parseDocument(source);
-    return document.toJSON() || {};
-  } catch (e) {
-    logger.error(`Failed to load index configuration from: ${configfile}`, e);
-    // config not usable return empty
-    return {
-      indices: [],
-    };
-  }
-}
-
-/**
- * Fetch HTML file from the file system.
- *
- * @param {object} params
- * @return HTML file
- */
-async function fetchHTMLFromFile(params) {
-  const { path, logger } = params;
-  const htmlfile = resolve(dirname(__dirname), path.replace(/\.md$/, '.html').substr(1));
-  logger.debug(`Reading HTML from: ${htmlfile}`);
-
-  return fs.readFile(htmlfile, 'utf8');
-}
-
-/**
- * Index loader and HTML fetcher that operates on the local file system.
- */
-const fileLoader = {
-  loadConfig: loadConfigFromFile,
-  fetchHTML: fetchHTMLFromFile,
-};
-
 /**
  * Load an index configuration from a git repo.
  *
  * @param {object} params
  * @returns configuration
  */
-async function loadConfigFromRepo(params) {
+async function loadConfig(params) {
```

Comment:
why not use `action.downloader` instead ?

Reply:
Right, I didn't notice that I was in a pipeline action. But that code didn't change, it dates back from when `action.downloader` wasn't yet available.

Reply:
but you changed so much in this PR.... where as the title says: use `helix-query.yaml instead of...`. for me, that would be a oneliner....

Reply:
Right, but as the initial comment on that PR says:
> Additionally, use nock to intercept HTTP requests for config and HTML files, instead of switching to a file-based loader in html_json.js just for testing.

After introducing `nock`, I was able to remove the file-based config and html loader (`loadConfigFromFile` and `fetchHTMLFromFile`), so e.g. `loadConfigFromRepo` now just becomes `loadConfig`. The function bodies didn't change.

Reply:
but anyways...it might be good to use the downloader now :-) or in a separate PR....